src/main/resources/jupyter/init-actions.sh

```diff
@@ -228,6 +228,9 @@ if [[ "${ROLE}" == 'Master' ]]; then
     log 'Starting Jupyter Notebook...'
     docker exec -d ${JUPYTER_SERVER_NAME} ${JUPYTER_SCRIPTS}/run-jupyter.sh
     log 'All done!'
+else
+    apt-get install -yq --no-install-recommends python3.5
```

Comment:
I wonder if you need to call `update_apt_get` in order for this to work. It looks like that is currently only being called on the master.